.github/workflows/Check.yml

```
@@ -0,0 +1,47 @@
+name: Build
+on:
+  schedule:
+  - cron: "5 1 * * *"
+  push:
+  workflow_dispatch:
+jobs:
+  build_latex:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Git repository
+        uses: actions/checkout@v2
+      - name: Compile LaTeX document (German)
+        uses: xu-cheng/texlive-action/full@v1
+        with:
+          run: |
+            ./compile.sh
+      - name: Compile LaTeX document (English)
+        uses: xu-cheng/texlive-action/full@v1
+        with:
+          run: |
+            ./compile.sh en
+      - name: Upload german PDF as Artifact
+        if: github.event_name == 'push'
+        uses: actions/upload-artifact@v2
+        with:
+          name: thesis_main.pdf
+          path:
+            thesis_main.pdf
+      - name: Upload english PDF as Artifact
+        if: github.event_name == 'push'
```

```suggestion
        if: github.ref != 'refs/heads/master' &&  github.event_name == 'push'
```
Not sure what the best way is to determine if the push happens in a PR, but adding the `ref` check here, avoids creating artifacts in master (because we create it on release).

---

That's a good Idea. Having the artifacts created would not be a real issue, but you are right, we don't need them for pull requests.